invenio_saml/handlers.py

```diff
@@ -18,6 +19,44 @@
 from .invenio_app import get_safe_redirect_target
 
 
+def account_info(attributes, remote_app):
+    """Return account info for remote user.
+
+    :param attributes: (dict) dictionary of data returned by identity provider.
+
+    :param remote_app: (str) Identity provider key.
+
+    :returns: (dict) A dictionary representing user to create or update.
+
+    :mappings extracts the mapping or attributes for given remote_app.
+
+    """
+    if current_app.config['SSO_SAML_IDPS']:
+        mappings = current_app.config['SSO_SAML_IDPS'][remote_app]['mappings']
+    else:
+        mappings = {
+            "email": "email",
+            "name": "name",
+            "surname": "surname",
+            "external_id": "external_id",
+        }
+
+    name = attributes[mappings['name']][0]
+    surname = attributes[mappings['surname']][0]
+    email = attributes[mappings['email']][0]
+    external_id = attributes[mappings['external_id']][0]
+
+    return dict(
+        user=dict(
+            email=email,
+            profile=dict(username=surname, full_name=name+" "+surname),
```

Comment:
here, I am not sure about the `username` profile, since its a unique field, there might be the case that there is already a similar username in the database saved.

Reply:
I just checked and at CERN we have a field `upn`: `User Principal Name, this is unique and can be used to identify a user.`.

Reply:
yes, that would be `external_id` - as we did in our institute something like:
external_id is something like: `77ABA62CDC768BF1@tugraz.at `, and we are assigning it to the username.
```
            profile=dict(username=external_id.split('@')[0],
```
would that work? and i am not sure that is the `@domain` at the end is also in your case

Reply:
For us, it does **not** have the domain name, for example we will have `mcurie`. We have an extra field `CernMailUpn` which is `mcurie@cern.ch`, so `[login]@cern.ch`.
The username should be without the `@`, so I think you can use the external id removing the domain.

Reply:
that's bad because we have the `@domain` at the end, and `@` is not something we can not add to the username.
 `Username must start with a letter, be at least three characters long and only contain alphanumeric characters, dashes and underscores.`  
 maybe I should add some check. or it would be better to combine `name + - + last name`?

Reply:
Can we do something like:
```
username  = external_id.split("@")[0] if "@" in external_id else external_id
```